### ISSUE TITLE
ansible-playbook fails, ansible works

### ISSUE THREAD

**Issue by elventear** [ansible/ansible#16845]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
ansible 2.2.0
  config file = /Path/To/ansible_vagrant.cfg
  configured module search path = Default w/o overrides
```

Based of `devel` commit aece03312f268fbdc268aa8c97541c583e605001
##### CONFIGURATION

My working `ansible.cfg`

```
[defaults]
inventory = ./vagrant_hosts
private_key_file = ./ssh/vagrant
hash_behaviour = merge
host_key_checking = False

[ssh_connection]
pipelining = True
ssh_args = -F /dev/null -C -o ControlMaster=auto -o ControlPersist=60s -o UserKnownHostsFile=/dev/null
```
##### OS / ENVIRONMENT

Connect from OS X.11.6 to Fedora 24, using MacPorts Python and SSH.
##### SUMMARY

I have a custom development setup, that works on Ansible 2.1.0. When upgrading to the `devel` it breaks. I have figured out that `ansible` still works, but `ansible-playbooks` is broken.
##### STEPS TO REPRODUCE

Run `ansible-playbooks` and it breaks right away. If I run `ansible` and it succeeds, `ansible-playbooks` will reuse the persisted SSH connection and succeed.

##### EXPECTED RESULTS

`ansible-playbooks` should succeed connecting every single time.
##### ACTUAL RESULTS

```
> ansible-playbook ../radia.yml -u root -vvvvv

Using /Path/To/vagrant/ansible_vagrant.cfg as config file
Loaded callback default of type stdout, v2.0

PLAYBOOK: radia.yml ************************************************************
3 plays in ../radia.yml

PLAY [all] *********************************************************************

TASK [bootstrap : install ansible minimum requirements] ************************
task path: /Path/To/roles/bootstrap/tasks/main.yml:2
<10.10.10.201> ESTABLISH SSH CONNECTION FOR USER: root
<10.10.10.200> ESTABLISH SSH CONNECTION FOR USER: root
<10.10.10.201> SSH: ansible.cfg set ssh_args: (-F)(/dev/null)(-C)(-o)(ControlMaster=auto)(-o)(ControlPersist=60s)(-o)(UserKnownHostsFile=/dev/null)
<10.10.10.201> SSH: ANSIBLE_HOST_KEY_CHECKING/host_key_checking disabled: (-o)(StrictHostKeyChecking=no)
<10.10.10.200> SSH: ansible.cfg set ssh_args: (-F)(/dev/null)(-C)(-o)(ControlMaster=auto)(-o)(ControlPersist=60s)(-o)(UserKnownHostsFile=/dev/null)
<10.10.10.201> SSH: ANSIBLE_PRIVATE_KEY_FILE/private_key_file/ansible_ssh_private_key_file set: (-o)(IdentityFile="./ssh/vagrant")
<10.10.10.200> SSH: ANSIBLE_HOST_KEY_CHECKING/host_key_checking disabled: (-o)(StrictHostKeyChecking=no)
<10.10.10.200> SSH: ANSIBLE_PRIVATE_KEY_FILE/private_key_file/ansible_ssh_private_key_file set: (-o)(IdentityFile="./ssh/vagrant")
<10.10.10.201> SSH: ansible_password/ansible_ssh_pass not set: (-o)(KbdInteractiveAuthentication=no)(-o)(PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey)(-o)(PasswordAuthentication=no)
<10.10.10.200> SSH: ansible_password/ansible_ssh_pass not set: (-o)(KbdInteractiveAuthentication=no)(-o)(PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey)(-o)(PasswordAuthentication=no)
<10.10.10.201> SSH: ANSIBLE_REMOTE_USER/remote_user/ansible_user/user/-u set: (-o)(User=root)
<10.10.10.200> SSH: ANSIBLE_REMOTE_USER/remote_user/ansible_user/user/-u set: (-o)(User=root)
<10.10.10.201> SSH: ANSIBLE_TIMEOUT/timeout set: (-o)(ConnectTimeout=10)
<10.10.10.200> SSH: ANSIBLE_TIMEOUT/timeout set: (-o)(ConnectTimeout=10)
<10.10.10.201> SSH: PlayContext set ssh_common_args: ()
<10.10.10.200> SSH: PlayContext set ssh_common_args: ()
<10.10.10.201> SSH: PlayContext set ssh_extra_args: ()
<10.10.10.200> SSH: PlayContext set ssh_extra_args: ()
<10.10.10.200> SSH: found only ControlPersist; added ControlPath: (-o)(ControlPath=/Path/To/.ansible/cp/ansible-ssh-%h-%p-%r)
<10.10.10.201> SSH: found only ControlPersist; added ControlPath: (-o)(ControlPath=/Path/To/.ansible/cp/ansible-ssh-%h-%p-%r)
<10.10.10.201> SSH: EXEC ssh -vvv -F /dev/null -C -o ControlMaster=auto -o ControlPersist=60s -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o 'IdentityFile="./ssh/vagrant"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Path/To/.ansible/cp/ansible-ssh-%h-%p-%r -tt 10.10.10.201 'dnf install -y python2 python2-dnf libselinux-python facter htop tmux'
<10.10.10.200> SSH: EXEC ssh -vvv -F /dev/null -C -o ControlMaster=auto -o ControlPersist=60s -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o 'IdentityFile="./ssh/vagrant"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Path/To/.ansible/cp/ansible-ssh-%h-%p-%r -tt 10.10.10.200 'dnf install -y python2 python2-dnf libselinux-python facter htop tmux'
fatal: [10.10.10.201]: UNREACHABLE! => {
    "changed": false,
    "msg": "Failed to connect to the host via ssh.",
    "unreachable": true
}
fatal: [10.10.10.200]: UNREACHABLE! => {
    "changed": false,
    "msg": "Failed to connect to the host via ssh.",
    "unreachable": true
}
    to retry, use: --limit @../radia.retry

PLAY RECAP *********************************************************************
10.10.10.200               : ok=0    changed=0    unreachable=1    failed=0
10.10.10.201               : ok=0    changed=0    unreachable=1    failed=0


> ansible -m ping all -u root -vvvvv
Using /Path/To/vagrant/ansible_vagrant.cfg as config file
Loaded callback minimal of type stdout, v2.0
Using module file /Path/To/vagrant/.venv/lib/python2.7/site-packages/ansible/modules/core/system/ping.py
<10.10.10.201> ESTABLISH SSH CONNECTION FOR USER: root
Using module file /Path/To/vagrant/.venv/lib/python2.7/site-packages/ansible/modules/core/system/ping.py
<10.10.10.201> SSH: ansible.cfg set ssh_args: (-F)(/dev/null)(-C)(-o)(ControlMaster=auto)(-o)(ControlPersist=60s)(-o)(UserKnownHostsFile=/dev/null)
<10.10.10.200> ESTABLISH SSH CONNECTION FOR USER: root
<10.10.10.201> SSH: ANSIBLE_HOST_KEY_CHECKING/host_key_checking disabled: (-o)(StrictHostKeyChecking=no)
<10.10.10.201> SSH: ANSIBLE_PRIVATE_KEY_FILE/private_key_file/ansible_ssh_private_key_file set: (-o)(IdentityFile="./ssh/vagrant")
<10.10.10.201> SSH: ansible_password/ansible_ssh_pass not set: (-o)(KbdInteractiveAuthentication=no)(-o)(PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey)(-o)(PasswordAuthentication=no)
<10.10.10.201> SSH: ANSIBLE_REMOTE_USER/remote_user/ansible_user/user/-u set: (-o)(User=root)
<10.10.10.201> SSH: ANSIBLE_TIMEOUT/timeout set: (-o)(ConnectTimeout=10)
<10.10.10.200> SSH: ansible.cfg set ssh_args: (-F)(/dev/null)(-C)(-o)(ControlMaster=auto)(-o)(ControlPersist=60s)(-o)(UserKnownHostsFile=/dev/null)
<10.10.10.201> SSH: PlayContext set ssh_common_args: ()
<10.10.10.200> SSH: ANSIBLE_HOST_KEY_CHECKING/host_key_checking disabled: (-o)(StrictHostKeyChecking=no)
<10.10.10.201> SSH: PlayContext set ssh_extra_args: ()
<10.10.10.200> SSH: ANSIBLE_PRIVATE_KEY_FILE/private_key_file/ansible_ssh_private_key_file set: (-o)(IdentityFile="./ssh/vagrant")
<10.10.10.200> SSH: ansible_password/ansible_ssh_pass not set: (-o)(KbdInteractiveAuthentication=no)(-o)(PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey)(-o)(PasswordAuthentication=no)
<10.10.10.200> SSH: ANSIBLE_REMOTE_USER/remote_user/ansible_user/user/-u set: (-o)(User=root)
<10.10.10.200> SSH: ANSIBLE_TIMEOUT/timeout set: (-o)(ConnectTimeout=10)
<10.10.10.201> SSH: found only ControlPersist; added ControlPath: (-o)(ControlPath=/Path/To/.ansible/cp/ansible-ssh-%h-%p-%r)
<10.10.10.200> SSH: PlayContext set ssh_common_args: ()
<10.10.10.200> SSH: PlayContext set ssh_extra_args: ()
<10.10.10.201> SSH: EXEC ssh -vvv -F /dev/null -C -o ControlMaster=auto -o ControlPersist=60s -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o 'IdentityFile="./ssh/vagrant"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Path/To/.ansible/cp/ansible-ssh-%h-%p-%r 10.10.10.201 '/bin/sh -c '"'"'/usr/bin/python && sleep 0'"'"''
<10.10.10.200> SSH: found only ControlPersist; added ControlPath: (-o)(ControlPath=/Path/To/.ansible/cp/ansible-ssh-%h-%p-%r)
<10.10.10.200> SSH: EXEC ssh -vvv -F /dev/null -C -o ControlMaster=auto -o ControlPersist=60s -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o 'IdentityFile="./ssh/vagrant"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Path/To/.ansible/cp/ansible-ssh-%h-%p-%r 10.10.10.200 '/bin/sh -c '"'"'/usr/bin/python && sleep 0'"'"''
10.10.10.201 | SUCCESS => {
    "changed": false,
    "invocation": {
        "module_args": {
            "data": null
        },
        "module_name": "ping"
    },
    "ping": "pong"
}
10.10.10.200 | SUCCESS => {
    "changed": false,
    "invocation": {
        "module_args": {
            "data": null
        },
        "module_name": "ping"
    },
    "ping": "pong"
}



```

**Comment by jctanner** [comment]
@elventear There doesn't appear to much a large difference between how ansible versus ansible-playbook attemts to connect to the machine, other than ansible-playbook uses -tt now.

Can you try running both of these commands and pasting the output?

```
ssh -vvv -F /dev/null -C -o ControlMaster=auto -o ControlPersist=60s -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o 'IdentityFile="./ssh/vagrant"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Path/To/.ansible/cp/ansible-ssh-%h-%p-%r -tt 10.10.10.200 'whoami'
ssh -vvv -F /dev/null -C -o ControlMaster=auto -o ControlPersist=60s -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o 'IdentityFile="./ssh/vagrant"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Path/To/.ansible/cp/ansible-ssh-%h-%p-%r 10.10.10.200 'whoami'
```

**Comment by ansibot** [comment]
@elventear ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dav1x** [comment]
Hey @jctanner 

I'm also having this issue. Do you want the full verbosity of the commands from above or just a summary?

**Comment by ansibot** [comment]
@elventear closure has been requested for this issue or it has timed out waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sdodson** [comment]
> @elventear closure has been requested for this issue or it has timed out waiting for your response.

Who/what requested closure? How do we request that it be reopened? bot help doesn't make it clear how to do that as a non member

**Comment by mscherer** [comment]
Reopening, since others people are ready to provides information. @dav1x , I guess that the full command would be what is needed, better have too much information than not enough for debugging.

**Comment by bcoca** [comment]
To clarify closure, the ticket was in 'needs_info' for many weeks and bot is now configured to close tickets that seem 'abandoned'.

**Comment by jctanner** [comment]
@mscherer @sdodson @dav1x the bot was looking for a response from the submitter. We're expecting the submitter to respond to info requests because they are the only person who can provide information for their specific environment. Many of these ssh related issues look similar but are completely different. It's very helpful when other community members provide data, but it doesn't always help resolve the same issue the submitter is reporting. 

So now that the ticket is open again and in needs info, the bot is going to close it again in 53 days unless the submitter responds. If you are not the submitter, you have a bug and are going to be be responsive, please open a new issue.

**Comment by elventear** [comment]
I apologize, but I do not have more time to spend on this issue. I have moved on and won't be able to revisit it. 

From my perspective, the bot was ok to close it.

**Comment by jctanner** [comment]
@elventear thank you for chiming in. I'll go ahead and re-close the issue.

**Comment by elventear** [comment]
It seems that with the latest release of `2.1.2.0`, this issue has surfaced on the stable branch, I have experienced it after upgrading and #17869 seems to sum up what is the problem. I hope this issue on `2.2` is also caused by the same problem.
